routers/web/repo/issue.go

```diff
@@ -2078,6 +2078,42 @@ func ViewIssue(ctx *context.Context) {
 		return user_service.CanBlockUser(ctx, ctx.Doer, blocker, blockee)
 	}
 
+	if ctx.IsSigned {
+		forkedRepos, err := repo_model.GetForksByUserAndOrgs(ctx, ctx.Doer, ctx.Repo.Repository)
+		if err != nil {
+			ctx.ServerError("GetForksByUserAndOrgs", err)
+			return
+		}
+		allowedRepos := make([]*repo_model.Repository, 0, len(forkedRepos)+1)
+		for _, repo := range append(forkedRepos, ctx.Repo.Repository) {
+			perm, err := access_model.GetUserRepoPermission(ctx, repo, ctx.Doer)
+			if err != nil {
+				ctx.ServerError("GetUserRepoPermission", err)
+				return
+			}
+			if perm.CanWrite(unit.TypeCode) {
+				allowedRepos = append(allowedRepos, repo)
+			}
+		}
+
+		ctx.Data["AllowedRepos"] = allowedRepos
+
+		devLinks, err := issue_service.FindIssueDevLinksByIssue(ctx, issue)
+		if err != nil {
+			ctx.ServerError("FindIssueDevLinksByIssue", err)
+			return
+		}
+		ctx.Data["DevLinks"] = devLinks
+		for _, link := range devLinks {
+			if link.LinkType == issues_model.IssueDevLinkTypePullRequest {
+				if !(link.PullRequest.Issue.IsClosed && !link.PullRequest.HasMerged) {
+					ctx.Data["MaybeFixed"] = link.PullRequest
```

Comment:
Will it be conflict with cross references? In some cases, the title may say "May be fixed by #1" and the comment list may say "User referenced a pull request that will close this issue (#2)".

Reply:
In future, I will add a new record in this table when there is a xref.